File: Leetcode/Invert Binary Tree/InvertBinaryTree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution 
{
    public TreeNode invertTree(TreeNode root) 
    {
        //empty tree
        if(root==null)                                          
            return root;
        
        //to traverse down left side of tree
        TreeNode left = invertTree(root.left);       
        
        //to traverse down right side of tree
        TreeNode right = invertTree(root.right);    
        
        //swap children
        root.right = left;                                      
        root.left = right;
        
        //return root of inverted binary tree
        return root;
    }
}

File: Leetcode/Invert Binary Tree/Invert Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    // Given the root of a binary tree, invert the tree, and return its root.
    TreeNode* invertTree(TreeNode* root) {
        if(root == NULL){
            return NULL;
        }
        
        TreeNode *temp = root->left;   // initializing a temporary struct. Could also use a pointer 
        root->left = root->right;    //swap children
        root->right = temp;
        invertTree(root->left);
        invertTree(root->right);
        return root;        //return root of inverted binary tree
       } 
};
